### PR TITLE
Use `builder-init` to generate project + run CI workflows.

### PR DESCRIPTION
**Pull request by ryan-roemer**
Parallel PR to: https://github.com/FormidableLabs/builder-init/pull/15

This adds a full project generation with `builder-init` to the root archetype CI. Once `builder-init` is released, I'll use the published version, rebase and release this PR.

/cc @coopy @benbayard @chaseadamsio @exogen @boygirl @zachhale @kenwheeler @jstrimpel @baer

### CODE REVIEW

File: init.js [hunk 1]
```diff
@@ -49,6 +62,10 @@ module.exports = {
   // Derived fields are asynchronous functions that are given the previous user
   // input data of the form: `function (data, cb)`. They callback with:
   // `(err, value)`.
+  //
+  // **Note**: We have implied derived values of:
+  // - `{{npmignore}}` -> `.npmignore`
+  // - `{{gitignore}}` -> `.gitignore`
```
Comment (rgerstenberger) [inline init.js]: I don't fully understand these lines, are these defaults?. Does this allow archetype authors to define `.npmignore` and `.gitignore` files?

Comment (ryan-roemer) [inline init.js]: See this full discussion https://github.com/FormidableLabs/builder-init/pull/15/files#diff-04c6e90faac2675aa89e2176d2eec7d8R224

**Comment by rgerstenberger** [comment]
Minor comment/question. Rest LGTM!

**Comment by ryan-roemer** [comment]
Release in `builder-react-component@0.2.0`

**Pull request merged**